python/tests/test_pca.py

```diff
@@ -379,7 +380,24 @@ def test_pca_spark_compat(
         model.setOutputCol("output")
         assert model.getOutputCol() == "output"
 
-        output = model.transform(df).collect()[0].output
+        k = model.getK()
+        output_df = model.transform(df)
+        schema_fields = [
+            StructField("features", VectorUDT(), True),
+        ]
+        if _PCA is SparkPCA:
+            schema_fields.append(
+                StructField(
+                    "output", VectorUDT(), True, metadata={"ml_attr": {"num_attrs": k}}
```

Comment:
Is this new?   Should we match this?

Reply:
No idea whether the metadata is now and how it is used. 

Seems not necessary to match it for now.